browser-tests/event-search/eventSearchPage.components.ts

```diff
@@ -0,0 +1,274 @@
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+import { screen, within } from '@testing-library/testcafe';
+import TestController from 'testcafe';
+
+import translations from '../../src/common/translation/i18n/fi.json';
+import { getEventFields } from '../../src/domain/event/EventUtils';
+import { KeywordOption } from '../../src/domain/event/types';
+import { formatDate } from '../../src/util/dateUtils';
+import getDateRangeStr from '../../src/util/getDateRangeStr';
+import toPascalCase from '../../src/util/toPascalCase';
+import { getErrorMessage } from '../utils/error.util';
+import { getExpectedEventContext } from '../utils/event.utils';
+import {
+  EventFieldsFragment,
+  Neighborhood,
+  PlaceFieldsFragment,
+} from '../utils/generated/graphql';
+
+export const getEventSearchPageComponents = (t: TestController) => {
+  const searchBanner = async () => {
+    await screen.findByTestId('searchContainer');
+    const selectors = {
+      searchBanner() {
+        return screen.findByTestId('searchContainer');
+      },
+      withinSearchBanner() {
+        t.ctx.withinTestId = 'searchContainer';
+        return within(screen.getByTestId(t.ctx.withinTestId));
+      },
+      searchButton() {
+        return this.withinSearchBanner().findByRole('button', { name: 'Hae' });
+      },
+      dateRangeFilter() {
+        return this.withinSearchBanner().findByLabelText('Valitse ajankohta');
+      },
+      dateRangeCheckbox(range: string) {
+        t.ctx.findByRole = ['checkbox', { name: range }];
+        return this.withinSearchBanner().findByRole.apply(
```

Comment:
Do we have some issue if we don't use .apply?

Reply:
Idea was to make shorter code, another option is:
```
        t.ctx.findByRole = { 
          matcher: 'checkbox', 
          options: { name: range } 
        };
        return this.withinSearchBanner().findByRole(
          t.ctx.findByRole.matcher,
          t.ctx.findByRole.options
        );
```
Good thing about ` t.ctx.findByRole ` is that it will tell exactly in error message what we tried to find if selector fails. testcafe's default error message does not tell the searched value `range` anywhere.

Reply:
Oh now I understand. Didn't rmember exact details how apply works, but I can see the benefit when now